twill/utils.py

```diff
@@ -128,6 +128,17 @@ def find_link(self, pattern: str) -> Optional[Link]:
             if regex.search(link.text) or regex.search(link.url):
                 return link
         return None
+    
+    def find_links(self, pattern: str) -> Optional[List[Link]]:
+        """Find all links with a given pattern on the result page."""
+        regex = re.compile(pattern)
+        links = []
+        for link in self.links:
+            if regex.search(link.text) or regex.search(link.url):
+                links.append(link)
+        if len(links) > 0:
```

Comment:
Is there a reason why you don't simply return an empty list in that case?

Reply:
Fixed